### PR TITLE
Repair create training data workflow

### PR DESCRIPTION
**Pull request by nbren12**
The create training data workflow was not working because of a few errors. See the commits for more info.

I also made some enhancements that simplify development.

* Minimize orchestration layers by moving templating into the docker image (scripts/run_single_training.sh) and doing the looping over months with argo rather than in the makefile.
* Improve docker caching by only copying the "scripts" directory.

### CODE REVIEW

File: projects/microphysics/create_training/Makefile [hunk 1]
```diff
@@ -1,24 +1,14 @@
...
-	./run_single.sh 07 $(CONFIG) $(OUTPUT_FREQUENCY) $(TAG_PREFIX)
-	./run_single.sh 08 $(CONFIG) $(OUTPUT_FREQUENCY) $(TAG_PREFIX)
-	./run_single.sh 09 $(CONFIG) $(OUTPUT_FREQUENCY) $(TAG_PREFIX)
-	./run_single.sh 10 $(CONFIG) $(OUTPUT_FREQUENCY) $(TAG_PREFIX)
-	./run_single.sh 11 $(CONFIG) $(OUTPUT_FREQUENCY) $(TAG_PREFIX)
-	./run_single.sh 12 $(CONFIG) $(OUTPUT_FREQUENCY) $(TAG_PREFIX)
+	argo submit argo.yaml \
+		-p config="$(shell base64 --wrap=0 $(CONFIG))" \
+		-p tag="${TAG_PREFIX}" \
+		-p output_frequency=${OUTPUT_FREQUENCY}
```
Comment (frodre) [inline projects/microphysics/create_training/Makefile:11]:
```suggestion
		-p tag="$(TAG_PREFIX)" \
		-p output_frequency="$(OUTPUT_FREQUENCY)"
```

File: projects/microphysics/create_training/argo.yaml [hunk 2]
```diff
@@ -46,18 +64,15 @@ spec:
...
-          --model NO_MODEL \
-          --job-type training-run \
-          --output-frequency "{{inputs.parameters.output_frequency}}" \
-          --offline
+      command:
+        - run_single_training.sh
+        - "{{workflow.parameters.config}}"
+        - "{{inputs.parameters.month}}"
+        - "{{workflow.parameters.tag}}"
+        - "{{workflow.parameters.output_frequency}}"
```
Comment (frodre) [inline projects/microphysics/create_training/argo.yaml:78]:
```suggestion
        - "{{workflow.parameters.output_frequency}}"

```

File: projects/microphysics/create_training/argo.yaml [hunk 3]
```diff
@@ -46,18 +64,15 @@ spec:
       - name: GOOGLE_APPLICATION_CREDENTIALS 
         value: /secret/gcp-credentials/key.json
       - name: WANDB_RUN_GROUP
-        value: "{{inputs.parameters.tag}}"
+        value: "{{workflows.parameters.tag}}"
+      - name: WANDB_JOB_TYPE
+        value: training-run
```
Comment (frodre) [inline projects/microphysics/create_training/argo.yaml:69]: This gets overwritten by the specification in `wandb.init`

File: projects/microphysics/scripts/prognostic_run.py [hunk 4]
```diff
@@ -61,9 +60,10 @@ def get_env(args):
 
 args = parser.parse_args()
 
-
 job = wandb.init(
-    job_type=args.job_type, project="microphysics-emulation", entity="ai2cm",
+    job_type=os.getenv("WANDB_JOB_TYPE", "prognostic_run"),
```
Comment (frodre) [inline projects/microphysics/scripts/prognostic_run.py:64]: Ah, nvm I see you updated.

File: projects/microphysics/scripts/run_single_training.sh [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+echo "Running the following configuration"
+cat fv3config.yaml
+
+prognostic_run.py \
+    --tag "$TAG-$MONTH" \
+    --model NO_MODEL \
+    --config-path fv3config.yaml \
+    --output-frequency "$OUTPUT_FREQUENCY" \
+    --offline
```
Comment (frodre) [inline projects/microphysics/scripts/run_single_training.sh:36]:
```suggestion
    --offline

```

**Review by frodre — APPROVED** [review]
LGTM, using environment variables makes things a little more complicated running from local but that was already the case with WANDB_RUN_GROUP.

**Comment by frodre** [comment]
Also curious what the main errors were?  Some updates in between merge of that PR or byproduct of me using locally built environments to test?

File: projects/microphysics/scripts/run_single_training.sh [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+export IC_URL="gs://vcm-ml-raw/2020-11-05-GFS-month-start-initial-conditions-year-2016/$TIMESTAMP"
+export MONTH_INT=$(printf '%d' $MONTH)
+envsubst < template.yaml > fv3config.yaml
+
+echo "Running the following configuration"
+cat fv3config.yaml
+
+prognostic_run.py \
+    --tag "$TAG-$MONTH" \
+    --model NO_MODEL \
```
Comment (frodre) [inline projects/microphysics/scripts/run_single_training.sh:33]: This could be left off since the default is "NO_MODEL" now

**Comment by nbren12** [comment]
1. `args.job_type` was used in the script, but it was called `args.wand_job_type` in the argument. see https://github.com/ai2cm/fv3net/pull/1516/commits/a3b395a38c47dbe20f387ef35c35e3291133c509
2.  non-existing output names in the configuration: c73e7f1

**Pull request merged**